src/libraries/System.Net.Primitives/src/System/Net/CookieContainer.cs

```diff
@@ -772,9 +773,9 @@ internal CookieCollection CookieCutter(Uri uri, string headerName, string setCoo
                 }
             }
 
-            foreach (Cookie c in cookies)
+            foreach (Cookie? c in cookies)
```

Comment:
Why is this needed?

I'll stop asking about this.  But CookieCollection prohibits null items from being stored in it, and it implements `IEnumerable<Cookie>` rather than `IEnumerable<Cookie?>`, so we shouldn't need these `?`s or associated `!`s.

Reply:
I guess because CookieCollection is defined as ICollection.
```c#
    public class CookieCollection : ICollection<Cookie>, IReadOnlyCollection<Cookie>, ICollection
```